pkg/manifest/ostree_deployment.go

```diff
@@ -193,51 +188,58 @@ func (p *OSTreeDeployment) serialize() osbuild.Pipeline {
 		)
 	}
 
-	pipeline.AddStage(osbuild.NewOSTreePullStage(
-		&osbuild.OSTreePullStageOptions{Repo: repoPath, Remote: p.Remote.Name},
-		osbuild.NewOstreePullStageInputs("org.osbuild.source", commit.Checksum, commit.Ref),
-	))
-
-	pipeline.AddStage(osbuild.NewOSTreeDeployStage(
-		&osbuild.OSTreeDeployStageOptions{
-			OsName: p.osName,
-			Ref:    commit.Ref,
-			Remote: p.Remote.Name,
-			Mounts: []string{"/boot", "/boot/efi"},
-			Rootfs: osbuild.Rootfs{
-				Label: "root",
+	var ref string
+	if p.ostreeSpec != nil {
+		commit := *p.ostreeSpec
+		ref = commit.Ref
+		pipeline.AddStage(osbuild.NewOSTreePullStage(
+			&osbuild.OSTreePullStageOptions{Repo: repoPath, Remote: p.Remote.Name},
+			osbuild.NewOstreePullStageInputs("org.osbuild.source", commit.Checksum, ref),
+		))
+
+		pipeline.AddStage(osbuild.NewOSTreeDeployStage(
+			&osbuild.OSTreeDeployStageOptions{
+				OsName: p.osName,
+				Ref:    ref,
+				Remote: p.Remote.Name,
+				Mounts: []string{"/boot", "/boot/efi"},
+				Rootfs: osbuild.Rootfs{
+					Label: "root",
+				},
+				KernelOpts: kernelOpts,
 			},
-			KernelOpts: kernelOpts,
-		},
-	))
+		))
 
-	remoteURL := p.Remote.URL
-	if remoteURL == "" {
-		// if the remote URL for the image is not specified, use the source commit URL
-		remoteURL = commit.URL
-	}
-	pipeline.AddStage(osbuild.NewOSTreeRemotesStage(
-		&osbuild.OSTreeRemotesStageOptions{
-			Repo: "/ostree/repo",
-			Remotes: []osbuild.OSTreeRemote{
-				{
-					Name:        p.Remote.Name,
-					URL:         remoteURL,
-					ContentURL:  p.Remote.ContentURL,
-					GPGKeyPaths: p.Remote.GPGKeyPaths,
+		remoteURL := p.Remote.URL
+		if remoteURL == "" {
+			// if the remote URL for the image is not specified, use the source commit URL
+			remoteURL = commit.URL
+		}
+		pipeline.AddStage(osbuild.NewOSTreeRemotesStage(
+			&osbuild.OSTreeRemotesStageOptions{
+				Repo: "/ostree/repo",
+				Remotes: []osbuild.OSTreeRemote{
+					{
+						Name:        p.Remote.Name,
+						URL:         remoteURL,
+						ContentURL:  p.Remote.ContentURL,
+						GPGKeyPaths: p.Remote.GPGKeyPaths,
+					},
 				},
 			},
-		},
-	))
+		))
 
-	pipeline.AddStage(osbuild.NewOSTreeFillvarStage(
-		&osbuild.OSTreeFillvarStageOptions{
-			Deployment: osbuild.OSTreeDeployment{
-				OSName: p.osName,
-				Ref:    commit.Ref,
+		pipeline.AddStage(osbuild.NewOSTreeFillvarStage(
+			&osbuild.OSTreeFillvarStageOptions{
+				Deployment: osbuild.OSTreeDeployment{
+					OSName: p.osName,
+					Ref:    ref,
+				},
 			},
-		},
-	))
+		))
```

Comment:
same for this one. we might not always want it.

Reply:
In what situation would we not need this?  We've always had it for deployments so I always considered it necessary.  The osbuild stage referenceshttps://github.com/coreos/fedora-coreos-config/blob/0b50084c9605d40c01c083e8b7b8efd9695d13a5/overlay.d/05core/usr/lib/dracut/modules.d/40ignition-ostree/ignition-ostree-populate-var.sh.

Reply:
well, for Fedora/RH CoreOS for instance, we have the file you linked that runs on boot and does those things, so we don't need them in the disk image.

Reply:
This stage isn't run when building the CoreOS image.  Resolving.